snap/snapcraft.yaml

```diff
@@ -231,33 +232,47 @@ parts:
   kafka:
     plugin: nil
     source: https://github.com/canonical/central-uploader/releases/download/kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}-ubuntu0/kafka_2.13-${SNAPCRAFT_PROJECT_VERSION}-ubuntu0-20231121145749.tgz
+    build-packages:
+    - ca-certificates-java
     stage-packages:
     - openjdk-17-jre-headless
     - util-linux
     - libpsl5
     - curl
     override-build: |
       snapcraftctl build
-
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
-
       cp -r ./* $SNAPCRAFT_PART_INSTALL/opt/kafka
-    override-prime: |
-      snapcraftctl prime
-      rm -vf usr/lib/jvm/java-17-openjdk-amd64/lib/security/cacerts
-  prometheus-exporter:
+  extra-maven-libs:
     plugin: nil
     after: [kafka]
     override-build: |
-      curl -o jmx_prometheus_javaagent.jar "https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.17.0/jmx_prometheus_javaagent-0.17.0.jar"
-      EXPECTED_SHA="48ab23a7514f9de5d2ca0acbb8ed1573b3c2ecbef7c5dc4d37c4ba125e538199"
-      JAR_SHA=$(sha256sum jmx_prometheus_javaagent.jar | cut -d " " -f1)
-      if [ $JAR_SHA != $EXPECTED_SHA ]; then exit 1; fi
-      cp jmx_prometheus_javaagent.jar $SNAPCRAFT_PART_INSTALL/
-    organize:
-      jmx_prometheus_javaagent.jar: opt/kafka/libs/jmx_prometheus_javaagent.jar
-
+      # path jar version symlink
+      jars=(
+        "io/strimzi kafka-oauth-common 0.14.0 false"
+        "io/strimzi kafka-oauth-client 0.14.0 false"
+        "io/strimzi kafka-oauth-server 0.14.0 false"
+        "io/strimzi kafka-oauth-server-plain 0.14.0 false"
+        "com/nimbusds nimbus-jose-jwt 9.37.1 false"
+        "io/prometheus/jmx jmx_prometheus_javaagent 0.17.0 true"
+      )
+      for entry in "${jars[@]}"; do
+        base=https://repo1.maven.org/maven2
+        path=$(echo "$entry" | awk '{print $1}')
+        jar_name=$(echo "$entry" | awk '{print $2}')
+        version=$(echo "$entry" | awk '{print $3}')
+        symlink=$(echo "$entry" | awk '{print $4}')
+        jar_file="$jar_name-$version.jar"
+        url="$base/$path/$jar_name/$version/$jar_file"
+        curl -O "$url"
+        expected_sha=$(curl "$url.sha1")
+        actual_sha=$(sha1sum "$jar_file" | awk '{print $1}')
+        [ $actual_sha != $expected_sha ] && exit 1
+        $symlink && ln -rs $jar_file $jar_name.jar
```

Comment:
**question** why are we symlinking JMX? I don't think we were symlinking this before.

Reply:
This was for the sake of keeping the names of jar files consistent, since every other jar has the version in the name. So the symlink was to avoid renaming (keep the jar version in the name), but I don't have a strong opinion on this. Do you want me to update it?

Reply:
Oh I see. You mean that we used to name the JMX file without providing the version. Actually, I believe it would be much better to have the JAR with the version there as well, as you are doing for the others. So, really, NO need to create the symlink, I would find it confusing actually.

Reply:
@marcoppenheimer @zmraul  I'm just thinking that it is kinda weird to have a symlink that refers to another jar (and therefore effectively the same file twice) in a class path. I'm wondering whether this creates some issue in Java. 

Maybe the best solution would be to handle this in the charm, where we auto-discover the version and we use the file in the properties after, or using an env-variable.

Reply:
I am resolving this conversation here, so we can continue it in the comment about splitting jmx from strimzi, since the two things are related.